### PR TITLE
Update to K2 + modernize

### PR DESCRIPTION
**Pull request by ZacSweers**
- Kotlin 2.0.20
- MoshiX 0.28.0
- JDK 22
- Gradle 8.10

### CODE REVIEW

**Comment by ZacSweers** [comment]
Exposed https://github.com/google/ksp/issues/1713 when trying KSP2

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -1,25 +1,25 @@
 [versions]
-kotlin = "1.9.24"
+kotlin = "2.0.0"
 kotlinx-serialization = "1.6.3"
-ksp = "1.9.24-1.0.20"
+ksp = "2.0.0-1.0.21"
 ktfmt = "0.49"
 jvmTarget = "17"
-moshix = "0.26.0"
+moshix = "0.27.0-alpha01"
```
Comment (ZacSweers) [inline gradle/libs.versions.toml:7]:
```suggestion
moshix = "0.27.1"
```

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -1,25 +1,25 @@
 [versions]
-kotlin = "1.9.24"
+kotlin = "2.0.0"
 kotlinx-serialization = "1.6.3"
-ksp = "1.9.24-1.0.20"
+ksp = "2.0.0-1.0.21"
```
Comment (ZacSweers) [inline gradle/libs.versions.toml:4]:
```suggestion
ksp = "2.0.0-1.0.22"
```

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -1,25 +1,25 @@
 [versions]
-kotlin = "1.9.24"
+kotlin = "2.0.0"
 kotlinx-serialization = "1.6.3"
```
Comment (ZacSweers) [inline gradle/libs.versions.toml:3]:
```suggestion
kotlinx-serialization = "1.7.0"
```

**Review by jstewart5000 — APPROVED** [review]

**Review by valeraz — APPROVED** [review]

**Pull request merged**